Mage/src/main/java/mage/cards/ExpansionSet.java

```diff
@@ -142,6 +142,7 @@ public static ExpansionSetComparator getComparator() {
     protected boolean hasOnlyMulticolorCards = false;
     protected boolean hasAlternateBoosterPrintings = true; // not counting basic lands; e.g. Fallen Empires true, but Tenth Edition false
 
+    protected int expectedMaxCardNumber; // used by Verify check to find unimplemented cards. Does not drive any decision on the set.
```

Comment:
1. Card viewer (`MageBook`) shows implemented/unimplemented stats per set. Current version search max card numbers by cards cycle, but now it can use that stats for more precise.
2. `test_checkWrongCardsDataInSets` can be improved with new check:
    * `card.getCardNumber() <= set.getExpectedMaxCardNumber()`

Reply:
3. If mtgjson has that info then `test_checkMissingSetData` it can be improved with new check about max card number

Reply:
Those are good uses for that number, I'll make a following issue to use them there and make sure that doesn't break any existing logic.
Also it then would make sense to add that number to more of the SetExpansion (I skipped a lot of reprint-only/preconstructed-deck sets). A few of them are not doable that way as they are not using card number the same ways.